src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp

```diff
@@ -780,6 +781,9 @@ void CompositionalMultiphaseBase::initializeFluidState( MeshLevel & mesh,
 
   } );
 
+  // with initial component densities defined - check if they need to be corrected to avoid zero diags etc
+  chopNegativeDensities( domain );
```

Comment:
I don't understand why we're getting negative component densities at the start. I'm guessing that means we either have negative total fluid density or negative component fractions both of which should not happen. I must be missing something?

Reply:
The function name is a bit confusing
What it does - trim the densities below minCompDens value (default is 1e-10)
This is important to avoid zero diags in Jacobian